src/status_im/contexts/shell/share/wallet/view.cljs

```diff
@@ -79,54 +82,54 @@
+  [rn/view {:style (wallet-style/indicator-wrapper-style active?)}])
 
 (defn- indicator-list
-  [indicator-count current-index]
-  [rn/view
-   {:style wallet-style/indicator-list-style}
-   (for [i (range indicator-count)]
-     (let [current-index (cond (<= current-index 0)                     0
-                               (>= current-index (dec indicator-count)) (dec indicator-count)
-                               :else                                    current-index)]
-       ^{:key i} [indicator (= current-index i)]))])
+  [num-indicators current-index]
+  [rn/view {:style wallet-style/indicator-list-style}
+   (for [i (range num-indicators)]
+     ^{:key i} [indicator (= current-index i)])])
 
 (defn render-item
   [item]
-  (let [width (rf/sub [:dimensions/window-width])]
-    [wallet-qr-code-item
-     {:account item
-      :index   (:position item)
-      :width   width}]))
+  [wallet-qr-code-item
+   {:account item
+    :index   (:position item)}])
+
+(defn- qr-code-visualized-index
+  [offset qr-code-size num-qr-codes]
+  (-> (+ (/ offset qr-code-size) 0.5)
+      (int)
+      (utils.number/value-in-range 0 (dec num-qr-codes))))
 
 (defn wallet-tab
   []
-  (let [accounts      (rf/sub [:wallet/accounts])
-        width         (rf/sub [:dimensions/window-width])
-        current-index (reagent/atom 0)]
+  (let [current-index         (reagent/atom 0)
+        {window-width :width} (rn/get-window)
+        qr-code-size          (- window-width 30)]
     (fn []
-      [rn/view
-       [rn/flat-list
-        {:horizontal                        true
-         :deceleration-rate                 0.9
-         :snap-to-alignment                 :start
-         :snap-to-interval                  (- width 30)
-         :disable-interval-momentum         true
-         :scroll-event-throttle             64
-         :data                              accounts
-         :directional-lock-enabled          true
-         :shows-horizontal-scroll-indicator false
-         :on-scroll                         (fn [e]
-                                              (reset! current-index (js/Math.ceil
-                                                                     (/ e.nativeEvent.contentOffset.x
-                                                                        width))))
-         :render-fn                         render-item}]
-       (when (> (count accounts) 1)
-         [rn/view
-          {:style {:margin-top 20}}
-          (indicator-list (count accounts) @current-index)])])))
+      (let [accounts     (rf/sub [:wallet/accounts])
+            num-accounts (count accounts)
+            on-scroll    (rn/use-callback
+                          (fn [e]
+                            (let [offset-x (oops/oget e "nativeEvent.contentOffset.x")
+                                  index    (qr-code-visualized-index offset-x qr-code-size num-accounts)]
+                              (reset! current-index index))))]
```

Comment:
This function works to calculate the highlighted dot below the QR codes, it has been changed to properly return the focused dot,